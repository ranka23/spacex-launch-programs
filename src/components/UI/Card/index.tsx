/* eslint-disable react/require-default-props */
import React, { ReactNode } from "react";
import styles from "./styles.module.scss";

interface CardProps {
  children: ReactNode;
  itemId?: string | number | null | undefined;
}

const Card: React.FC<CardProps> = ({ children, itemId }: CardProps) => {
  return (
    <div key={itemId} className={styles.card_container}>
      {children}
    </div>
  );
};
export default Card;
