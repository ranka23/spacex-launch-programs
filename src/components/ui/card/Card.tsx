/* eslint-disable react/require-default-props */
import React from "react";
import styles from "./card.module.scss";

interface CardProps {
  children: JSX.Element[];
  itemId?: string | number | null | undefined;
}

const Card: React.FC<CardProps> = ({ children, itemId }: CardProps) => {
  return (
    <div key={itemId} className={styles.card_container}>
      {children}
    </div>
  );
};
export default Card;
