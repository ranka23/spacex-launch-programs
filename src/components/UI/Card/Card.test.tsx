import React, { ReactNode } from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import Card from "./index";

describe("<Card />", () => {
  const tree = (children?: ReactNode) =>
    render(
      <MemoryRouter>
        <Card>{children}</Card>
      </MemoryRouter>
    ).container.firstChild;

  it("Should render nothing if no children", () => {
    expect(tree()).toMatchSnapshot();
  });

  it("Should render children if no errors", () => {
    expect(
      tree(
        <div>
          <h1>Hello World</h1>
        </div>
      )
    ).toMatchSnapshot();
  });
});
