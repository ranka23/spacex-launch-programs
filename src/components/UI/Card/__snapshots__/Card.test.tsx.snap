// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Should render children if no errors 1`] = `
<div
  class="card_container"
>
  <div>
    <h1>
      Hello World
    </h1>
  </div>
</div>
`;

exports[`<Card /> Should render nothing if no children 1`] = `
<div
  class="card_container"
/>
`;
